tests/phpunit/tests/general/wpPreloadLinks.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+/**
+ * @group general
+ * @group template
+ * @ticket 42438
+ * @covers ::wp_preload_links
+ */
+class Tests_General_wpPreloadLinks extends WP_UnitTestCase {
+
+	/**
+	 * @dataProvider provider_preload_links
+	 */
+	public function test_preload_links( $expected, $urls ) {
+		$callback = function () use ( $urls ) {
```

Comment:
As above, seems `$urls` is not a good descriptive name for this var. It doesn't contain just URLs, right? `link_attributes` would probably be better.

Reply:
renamed `preload_resources`